### PR TITLE
Nnc location

### PR DESCRIPTION
**Pull request by joakim-hove**
This work started out to improve the error messages emitted while processing the `NNC` and `EDITNNC` keywords. Since these keywords are processed in three modules, opm-common, opm-grid and opm-simulators, this was a bit more involved than the normal "improve the error messages" job. As part of the job the opm-common input processing of `NNC` and `EDITNNC` has been refactored quite a bit, these changes also affect downstream. The changes include:

1. Both the `NNC` and `EDITNNC` keyword are handled in the same class `NNC`.
2. When the same NNC is mentioned both in  the `NNC` and `EDITNNC` keywords the transmissibility scaling is handled in opm-common.
3. The `NNC/EDITNNC`entries  are sorted. 

Downstream: 
https://github.com/OPM/opm-grid/pull/499
https://github.com/OPM/opm-simulators/pull/2907

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

File: src/opm/parser/eclipse/EclipseState/Grid/NNC.cpp [hunk 1]
```diff
@@ -63,14 +63,6 @@ namespace Opm
         return result;
     }
 
-    void NNC::addNNC(const size_t cell1, const size_t cell2, const double trans) {
```
Comment (GitPaean) [inline src/opm/parser/eclipse/EclipseState/Grid/NNC.cpp:66]: Hi, I need this function to append more NNCs from numerical aquifers. Maybe I can add it back.

Comment (joakim-hove) [reply src/opm/parser/eclipse/EclipseState/Grid/NNC.cpp]: OK - I will leave it in for now; do you have a work in progress branch for the numerical aquifer work which you could point me to?

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this opm-simulators=2907 please

**Comment by joakim-hove** [comment]
jenkins build this opm-simulators=2907 please

**Comment by joakim-hove** [comment]
jenkins build this opm-simulators=2907 please

File: src/opm/parser/eclipse/EclipseState/Grid/NNC.cpp (hunk 1 above)
Comment (GitPaean) [reply src/opm/parser/eclipse/EclipseState/Grid/NNC.cpp:66]: Yes. Most likely I am creating a WIP PR today.  The branch is https://github.com/GitPaean/opm-common/tree/numerical_aquifer

**Comment by joakim-hove** [comment]
jenkins build this opm-simulators=2907 please

**Comment by joakim-hove** [comment]
jenkins build this opm-grid=499 opm-simulators=2907 please

**Comment by joakim-hove** [comment]
jenkins build this opm-grid=499 opm-simulators=2907 please

**Comment by joakim-hove** [comment]
jenkins build this opm-grid=499 opm-simulators=2907 please

**Comment by blattms** [comment]
Cool. I might be missing something, but it seems we are not checking `nnc.cell1 <=nnc.cell2` in the tests, are we?

**Comment by joakim-hove** [comment]
> I might be missing something, but it seems we are not checking nnc.cell1 <=nnc.cell2 in the

That is right - I am actually just working on it.

**Comment by joakim-hove** [comment]
jenkins build this opm-grid=499 opm-simulators=2907 please

**Comment by joakim-hove** [comment]
jenkins build this opm-grid=499 opm-simulators=2907 please

**Comment by joakim-hove** [comment]
jenkins build this opm-grid=499 opm-simulators=2907 please

**Comment by joakim-hove** [comment]
jenkins build this opm-grid=499 opm-simulators=2907 please

**Comment by joakim-hove** [comment]
jenkins build this opm-grid=499 opm-simulators=2907 please

**Comment by joakim-hove** [comment]
Comments to this?

**Comment by joakim-hove** [comment]
jenkins build this opm-grid=499 opm-simulators=2907 please

**Comment by joakim-hove** [comment]
jenkins build this opm-grid=499 opm-simulators=2907 please

**Pull request merged**